pkg/stub/imagestreams.go

```diff
@@ -221,41 +221,44 @@ func (h *Handler) upsertImageStream(imagestreamInOperatorImage, imagestreamInClu
 }
 
 func (h *Handler) updateDockerPullSpec(oldies []string, imagestream *imagev1.ImageStream, opcfg *v1.Config) {
-	//TODO remove when TBR creds sorted out and we do not explicitly set SamplesRegistry to registry.access.redhat.com
-	// by default we want to leave the jenkins images as using the payload set to the IMAGE* env's
-	// but if the customer overrides (after we switch to TBR and don't set SamplesRegistry by default), we'll let them
-	// change jenkins related images
-	switch imagestream.Name {
-	case "jenkins":
-		return
-	case "jenkins-agent-nodejs":
-		return
-	case "jenkins-agent-maven":
-		return
-	}
-	if len(opcfg.Spec.SamplesRegistry) > 0 {
-		logrus.Debugf("updateDockerPullSpec stream %s has repo %s", imagestream.Name, imagestream.Spec.DockerImageRepository)
-		// don't mess with deprecated field unless it is actually set with something
-		if len(imagestream.Spec.DockerImageRepository) > 0 &&
-			!strings.HasPrefix(imagestream.Spec.DockerImageRepository, opcfg.Spec.SamplesRegistry) {
-			// if not one of our 4 defaults ...
-			imagestream.Spec.DockerImageRepository = h.coreUpdateDockerPullSpec(imagestream.Spec.DockerImageRepository,
-				opcfg.Spec.SamplesRegistry,
-				oldies)
+	registryReplacement := "registry.redhat.io"
+	if len(opcfg.Spec.SamplesRegistry) == 0 {
+		// in our default case we want to leave the jenkins images as using the payload set to the IMAGE* env's;
+		// but if they explicitly set SamplesRegistry to registry.redhat.io, we'll let them explicity point to there
+		// in case the payload ever pointed to something different (though that is not the expectation for 4.0)
+		switch imagestream.Name {
+		case "jenkins":
+			return
+		case "jenkins-agent-nodejs":
+			return
+		case "jenkins-agent-maven":
+			return
```

Comment:
this doesn't seem to do what the comment says.... it looks like it'll only leave the jenkins imagestreams alone if no registry was specified.  But it says the intent is to leave the jenkins imagestreams alone unless the registry is explicitly set to registry.redhat.io.

In any case my personal take is we should just never touch the jenkins imagestreams.  They're part of the payload, so the users don't get to control where it should be imported from.  If we ever realize that's a mistake, we can introduce new behavior in an upgrade :)

Reply:
yeah ... and the can always mark them in the skip list if they want to change them ... will change

Reply:
part of next push